includes/class-wc-payments-account.php

```diff
@@ -184,7 +184,7 @@ public function is_stripe_connected( bool $on_error = false ): bool {
 	public function try_is_stripe_connected(): bool {
 		$account = $this->get_cached_account_data();
 		if ( false === $account ) {
-			throw new Exception( __( 'Failed to detect connection status', 'woocommerce-payments' ) );
+			throw new Exception( esc_html__( 'Failed to detect connection status', 'woocommerce-payments' ) );
```

Comment:
Thanks for working on this quickly, but I wonder if HTML-safe escaping is really required here. As per the [method's docs](https://developer.wordpress.org/reference/functions/esc_html__/), doing so is useful when the translated string will be embedded in the page markup.

Reply:
Thanks for looking!

Since it doesn't harm to wrap it in. `esc_html`, I was following @marcinbot's suggestion on Slack: p1711452617673489/1711443600.104489-slack-CGGCLBN58

I think the message is one which _would_ be safe to return on the front-end if we chose to, so it seems approprate to wrap it. 

In fact I seem to remember seeing that message returned during the account connection in the past (although I've looked at the code to try and find where, so maybe it was my imagination! 😅 )

Reply:
Gotcha, sounds good.